### ISSUE TITLE
Add support for custom labels for RootSyncs and RepoSyncs

### ISSUE THREAD

**Issue by samirtahir91** [GoogleContainerTools/kpt-config-sync#1380]
Hi

Is there plan to add support for additional labels for RootSyncs and RepoSyncs that will reflect in their respective reconiler deployments?  If not, can this be added please?

Thanks 
Samir

**Comment by sdowell** [comment]
This is not currently planned. Could you provide a bit more detail of the behavior you are looking for and some example use cases?

**Comment by samirtahir91** [comment]
For example, in my org they like to label every workload so that costing can be billed to departments/teams internally.

**Comment by mikebz** [comment]
@samirtahir91 is this billing for the workloads that Config Sync creates or Config Sync itself?

**Comment by samirtahir91** [comment]
> @samirtahir91 is this billing for the workloads that Config Sync creates or Config Sync itself?

The actual Root/RepoSync deployments, since there will be many in a large multi-tenant cluster scenario.

**Comment by janetkuo** [comment]
@samirtahir91 I assume you'd want to provide team information in Root/RepoSync CR objects, correct? Would annotations instead of labels work for your case? Is labeling Deployments enough, or do you actually need to label pods?

This could potentially be supported by copying all user-specified labels from Root/RepoSync objects to their corresponding reconciler Deployments. For consistency, we might consider propagating labels to other resources that reconciler manager creates (such as service accounts) too. 

Given that labeling affects how controllers select objects, this requires some design to make sure label propagation won't cause any side effects.

**Comment by samirtahir91** [comment]
@janetkuo Correct, we need some way to apply labels we want on the objects created/used for reconciling in the Root/RepoSync objects themselves, in my org we use labels only.

**Comment by samirtahir91** [comment]
We'd also want to add custom annotations as well as labels.
